src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -996,6 +1358,16 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
         md->flags |= UCT_IB_MLX5_MD_FLAG_MP_XRQ_FIRST_MSG;
     }
 
+    status = uct_ib_mlx5dv_check_dc(md);
```

Comment:
why do we need it given the check on line 1300?

Reply:
so you want to get dc ratio rather than check for DC availability? I think there is no need to re-evalute dc support by this dv function. Also we enable TM_DC support on the line 1304 and do no disable it even if this function fails (even though it is not supposed to happen)

Reply:
> so you want to get dc ratio rather than check for DC availability? 

right, it gets DC WQE ratio.

> I think there is no need to re-evalute dc support by this dv function.

this function check DC availability and calculates DC WQE ratio - unfirtunately, we can't split it

Reply:
> i think we need to refactor it somehow. Otherwise this code allows to disable DC capability, but keep DC TM enabled

It won't disable, it adds `UCT_IB_DEVICE_FLAG_DC` flag.
Maybe is it worth checking which type of DCI is going to be used (DEVX or DV) and use the correct method of validation whether it is supported or not?

Reply:
smth like:
```
```